api/src/main/java/org/xwiki/apps/googleapps/GoogleAppsManager.java

```diff
@@ -0,0 +1,242 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.apps.googleapps;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.services.drive.model.File;
+import com.google.api.services.drive.model.FileList;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * The specification of the methods that the manager of the GoogleApps application
+ * is doing. Methods of this interface are mostly called by the script-service (itself
+ * called by the views).
+ *
+ * @version $Id$
+ * @since 3.0
+ */
+@Role
+public interface GoogleAppsManager
+{
+
+
+    /**
+     * @return if the application is licensed and activated
+     * @throws XWikiException in case a context cannot be read from thread.
+     * @since 3.0
+     */
+    @Unstable
+    boolean isActive() throws XWikiException;
+
+    /**
+     *
+     * @return if the app is configured to use the Google Drive integration (default: yes).
+     * @since 3.0
+     */
+    @Unstable
+    boolean isUseDrive();
+
+
+    /**
+     * Reads the manifest to find when the JAR file was assembled by maven.
+     * @return the build date.
+     * @since 3.0
+     */
+    @Unstable
+    Date getBuildTime();
+
+
+    /** Inspects the stored information to see if an authorization or a redirect needs to be pronounced.
+     *
+     * @return found credential
+     * @throws XWikiException if the interaction with xwiki failed
+     * @throws IOException if a communication problem to Google services occured
+     * @since 3.0
+     */
+    @Unstable
+    Credential authorize() throws XWikiException, IOException;
+
+
+    /** Inspects the stored information to see if an authorization or a redirect needs to be pronounced.
+     *
+     * @param redirect If a redirect can be done
+     * @return found credential
+     * @throws XWikiException if the interaction with xwiki failed
+     * @throws IOException if a communication problem to Google services occured
+     * @since 3.0
+     */
+    @Unstable
+    Credential authorize(boolean redirect) throws XWikiException, IOException;
+
+    /**
+     * Performs the necessary communication with Google-Services to fetch identity and
+     * update the XWiki-user object or possibly sends a redirect to a Google login screen.
+     *
+     * @return "failed login" if failed, "no user" (can be attempted to Google-OAuth),
+     *          or "ok" if successful
```

Comment:
We need the different strings because we need the view to be properly forking. It was a number before.
Adding an explanatory message based on some exception in the communication with Google could be done later but we need to unwrap things first. This is not porting anymore but enriching. 
I suggest to postpone this. Ok?

Reply:
OK.